### PR TITLE
chore(docs): add workflow for synchronizing docs with docs.konghq.com

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

It adds a workflow that synchronize CRD's auto-generated reference with docs.konghq.com repository. It does some post-processing required for docs only (adding a title, turning vale linting off).

Typically the workflow will be triggered manually on every release, but it can also be used when docs synchronization will be required between releases (caution has to be taken in order to not publish API changes that haven't been released yet).

**Which issue this PR fixes**:

Part of #3019.

### CODE REVIEW

File: scripts/apidocs-gen/post-process-for-konghq.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#   - adds a title section
+#   - turns vale linter off for the whole document
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")/../.."
+CRD_REF_DOC="${SCRIPT_ROOT}/docs/api-reference.md"
+POST_PROCESSED_DOC="$1"
```
Comment (pmalek) [inline scripts/apidocs-gen/post-process-for-konghq.sh:14]:
```suggestion
POST_PROCESSED_DOC="${1}"
```

**Review by pmalek — COMMENTED** [review]
Now that I think of it: we really shouldn't update that more often than "on a release" because we'd ship to users docs on something  that is not yet available to them, right?

**Comment by czeslavo** [comment]
> Now that I think of it: we really shouldn't update that more often than "on a release" because we'd ship to users docs on something that is not yet available to them, right?

Hmm. Yes, you're right. We shouldn't publish API changes before the release.

On the other hand, making this a step of the release pipeline can block us if we decided to fix the docs themselves (with no API changes) in the meantime. That still could be mitigated by changing godocs, generating markdown, and excluding upcoming release changes manually. Probably shouldn't happen that often. 🤔 WDYT?

**Comment by pmalek** [comment]
> On the other hand, making this a step of the release pipeline can block us if we decided to fix the docs themselves (with no API changes) in the meantime

Can you elaborate on this? I'm not sure I understand the problem here.

**Comment by czeslavo** [comment]
> > On the other hand, making this a step of the release pipeline can block us if we decided to fix the docs themselves (with no API changes) in the meantime
> 
> Can you elaborate on this? I'm not sure I understand the problem here.

I mean the following situation:
1. We're before a release `X+1`, some API changes have been made. 
2. We've noticed that something is missing in the docs released for `X`.
3. We want to fix the docs for release `X`, but we shouldn't publish parts that are not released yet in `X+1`. 

In such case, we would have to change godocs manually, generate the markdown, and copy it to the docs.konghq.com manually, excluding consciously pieces that are going to be released in `X+1`.

**Comment by czeslavo** [comment]
> Got it. In that case why not make this workflow also possible to be triggered manually?
Yeah, I think it will make sense. I will just make sure that the PR will be created as a draft so a person triggering it will have to verify whether we're not publishing something that shouldn't be published yet.

**Comment by pmalek** [comment]
> > Got it. In that case why not make this workflow also possible to be triggered manually?
> 
> Yeah, I think it will make sense. I will just make sure that the PR will be created as a draft so a person triggering it will have to verify whether we're not publishing something that shouldn't be published yet.

The only problem with this that I see is that in a situation when we already change something for release `N+1` and at that point we'd notice something broken in current release `N`, we'd have no way of making this "a pleasant experience" (without manually removing some sections etc.).

Please also be sure to include this in https://github.com/Kong/kubernetes-ingress-controller/blob/main/RELEASE.md

**Comment by czeslavo** [comment]
@pmalek I made it a separate workflow so it can be triggered manually at a suitable time. I also added a release issue template section for that. PTAL 🙇

**Review by pmalek — APPROVED** [review]

**Pull request merged**